### ISSUE TITLE
[studio-ui]  TinyMCE5 feature to switch between code editor and RTE in same DOM

### ISSUE THREAD

**Issue by phuongnq** [craftercms/craftercms#4894]
# Feature Request

Crafter CMS 3.1.x latest comes with TinyMCE5 as the default editor. In previous versions, users can switch to HTML code editor within the RTE view area. But in TinyMCE5, the code editor is displayed in a popup. Is there a way to support switching to the source code editor within the RTE view area instead of a popup?

#### Is your feature request related to a problem? Please describe.

Due to the popup displaying, users cannot open multiple TinyMCE5 to copy from field to field.

#### Describe the solution you'd like

An option to switch source code editor and RTE within the same dom instead of a popup.

#### Describe alternatives you've considered

I created a plugin that can display both RTE and source code. But looking for a better solution if available.

Here is my plugin:

* Source code: https://github.com/phuongnq/craftercms-rte-plugin-code-view

* Sample video: 

https://user-images.githubusercontent.com/2996543/127279735-d6fda8f5-1ba9-4fcf-8c05-79fe54acea12.mp4

**Comment by sumerjabri** [comment]
@alhambrav @yacdaniel FYI

**Comment by jvega190** [comment]
@phuongnq assigning it to you for validation, then please assign @yacdaniel and @alhambrav

**Comment by phuongnq** [comment]
verified

**Comment by yacdaniel** [comment]
HI @jvega190 please check the usage of code editor in Preview(edit mode), version 4.0
When clicked `Tools - Code Editor` the toolbar is disabled. 
![image](https://user-images.githubusercontent.com/6722074/140335274-ab373bbe-bb3a-438e-b43b-270383858b43.png)


```
plugin.min.js:1 Uncaught DOMException: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.
    at u (http://ubuntu:8080/studio/static-assets/js/tinymce-plugins/ace/plugin.min.js:1:3100)
    at onAction (http://ubuntu:8080/studio/static-assets/js/tinymce-plugins/ace/plugin.min.js:1:2158)
    at http://ubuntu:8080/studio/static-assets/modules/editors/tinymce/v5/tinymce/themes/silver/theme.min.js:9:132108
    at http://ubuntu:8080/studio/static-assets/modules/editors/tinymce/v5/tinymce/themes/silver/theme.min.js:9:132646
    at http://ubuntu:8080/studio/static-assets/modules/editors/tinymce/v5/tinymce/themes/silver/theme.min.js:9:23058
    at mt (http://ubuntu:8080/studio/static-assets/modules/editors/tinymce/v5/tinymce/themes/silver/theme.min.js:9:3543)
    at run (http://ubuntu:8080/studio/static-assets/modules/editors/tinymce/v5/tinymce/themes/silver/theme.min.js:9:23035)
    at http://ubuntu:8080/studio/static-assets/modules/editors/tinymce/v5/tinymce/themes/silver/theme.min.js:9:30578
    at http://ubuntu:8080/studio/static-assets/modules/editors/tinymce/v5/tinymce/themes/silver/theme.min.js:9:412
    at http://ubuntu:8080/studio/static-assets/modules/editors/tinymce/v5/tinymce/themes/silver/theme.min.js:9:156924
 ```

**Comment by yacdaniel** [comment]
Verified fixed in 4.0.
